components/brave_ads/browser/BUILD.gn

```diff
@@ -42,38 +42,58 @@ source_set("browser") {
       "ads_service_impl.h",
       "background_helper.cc",
       "background_helper.h",
-      "background_helper_android.cc",
-      "background_helper_android.h",
-      "background_helper_linux.cc",
-      "background_helper_linux.h",
-      "background_helper_mac.mm",
-      "background_helper_mac.h",
-      "background_helper_win.cc",
-      "background_helper_win.h",
       "locale_helper.cc",
       "locale_helper.h",
-      "locale_helper_linux.cc",
-      "locale_helper_linux.h",
-      "locale_helper_mac.mm",
-      "locale_helper_mac.h",
-      "locale_helper_win.cc",
-      "locale_helper_win.h",
-      "locale_helper_android.cc",
-      "locale_helper_android.h",
       "bundle_state_database.cc",
       "bundle_state_database.h",
       "notification_helper.cc",
       "notification_helper.h",
-      "notification_helper_android.cc",
-      "notification_helper_android.h",
-      "notification_helper_linux.cc",
-      "notification_helper_linux.h",
-      "notification_helper_mac.mm",
-      "notification_helper_mac.h",
-      "notification_helper_win.cc",
-      "notification_helper_win.h",
     ]
 
+    if (is_android) {
```

Comment:
Was this a required change for the latest Chromium bump as previously the build toolchain only compiled the required files for each platform

Reply:
It's not required as of now yet, but we will have to do it eventually due to this effort at https://cs.chromium.org/chromium/src/docs/no_sources_assignment_filter.md
https://bugs.chromium.org/p/chromium/issues/detail?id=1018739